### ISSUE TITLE
[aws-rds] Supports defining variables for a CfnDBInstance in a DatabaseCluster

### ISSUE THREAD

**Issue by hixi-hyi** [aws/aws-cdk#9926]
Add the following properties in DatabaseCluster.
```
autoMinorVersionUpgrade: boolean
allowMajorVersionUpgrade: boolean
deleteAutomatedBackups: boolean
enablePerformanceInsights: boolean
```
The most important one is `autoMinorVersionUpgrade`.

#### `autoMinorVersionUpgrade`
When the database instance is automatically upgrade by AWS, drift is detected between cdk(cloudformation) and the actual resource, and the upgrade cannot be performed with cdk.
To work around this and deploy the cloudformation, you need to remove the RDS from the management target and re-include it. cdk does not support this.
We should be able to manage the migration in our own time.

#### Others (`allowMajorVersionUpgrade`, `deleteAutomatedBackups`, `enablePerformanceInsights`)
I'd like to set these parameters as well.

### Proposed Solution

1. Define the properties below
https://github.com/aws/aws-cdk/blob/3c116f3b5854726bef610e425fc35f2d2b50a179/packages/%40aws-cdk/aws-rds/lib/props.ts#L10
2. Pass properties to CfnDBInstance
https://github.com/aws/aws-cdk/blob/3c116f3b5854726bef610e425fc35f2d2b50a179/packages/%40aws-cdk/aws-rds/lib/cluster.ts#L590

### Other

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by theo-beers** [comment]
To add to this: we received some mails from AWS where they actually advised to disable autoMinorVersionUpgrade to prevent stack drift.

**Comment by hixi-hyi** [comment]
The `enablePerformanceInsights` is implemented by #10092
